### PR TITLE
32 Implement getMaxLeaf method

### LINKED ISSUE #32: Implement getMaxLeaf

### PR DESCRIPTION
**Pull request by RylanRCox**
I wrote two different methods, one is commented out because I'm not sure which one would work with what we have. One of them uses a getLeaves helper method that I made.

### CODE REVIEW

File: team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java [hunk 1]
```diff
@@ -90,13 +90,58 @@ private boolean playthrough(TreeNode current) {
...
+			if (leaves.get(i).getUCB() > check)
+				leafIndex = i;
+		}
+
+		//Returning the largest leaf.
+		return leaves.get(leafIndex);
+	}
+
+	private ArrayList<TreeNode> getLeaves(TreeNode root) {
+		//This is a helper class for get max leaf.
```
Comment (jpabadir) [inline team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java]: Minor: "method" instead of "class"

**Review by jpabadir — APPROVED** [review]
This looks good to me. I only reviewed the uncommented method. This approval assumes the commented-out method before merging.

File: team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java [hunk 2]
```diff
@@ -91,8 +91,38 @@ private boolean playthrough(TreeNode current) {
...
+
+		//Variables used to check against in the for loop
+		double check = leaves.get(0).getUCB();
+		int leafIndex = 0;
+
+		//Iterate through the array list of leaves finding the largest leaf.
+		for (int i = 1; i < leaves.size(); i++) {
+			if (leaves.get(i).getUCB() > check)
+				leafIndex = i;
+		}
```
Comment (JadenBalogh) [inline team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java:107]: I don't think this code works at a glance. The `check` will never update. Also, maybe rename `check` to something more descriptive (i.e. `maxUCB`).

File: team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java [hunk 3]
```diff
@@ -91,8 +91,38 @@ private boolean playthrough(TreeNode current) {
...
+
+	private ArrayList<TreeNode> getLeaves(TreeNode root) {
+		//This is a helper method for get max leaf.
+		ArrayList<TreeNode> leaves = new ArrayList();
+
+		for (int i = 0; i < root.children.size(); i++) {
+			//This if else statement checks if the child has children or is a leaf and then either adds it to the array list
+			if (root.children.get(i).children.isEmpty())
+				leaves.add(root.children.get(i));
+				//Or recursively calls itself to find more leaves.
```
Comment (JadenBalogh) [inline team-01/src/main/java/ubc/cosc322/MonteCarloPlayer.java:121]: This comment should be above the proper line.

**Review by JadenBalogh — CHANGES_REQUESTED** [review]

**Pull request merged**